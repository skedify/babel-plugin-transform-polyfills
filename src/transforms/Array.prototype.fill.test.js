import compile from '../../test/utils/compile'

describe('Array.prototype.fill', () => {
  it('should replace Array.prototype.fill with a polyfill', () => {
    const result = compile(
      `
      Array(10).fill()
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      Array(10).fill()
      Array(10).fill(2)
    `
    )

    expect(result).toMatchSnapshot()
  })
})
