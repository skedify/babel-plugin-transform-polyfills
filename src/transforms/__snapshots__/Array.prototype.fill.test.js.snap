// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array.prototype.fill should not import polyfill more than once 1`] = `
"import _Array_prototype_fill from \\"babel-plugin-transform-polyfills/polyfills/Array/prototype/fill\\";

_Array_prototype_fill(Array(10));

_Array_prototype_fill(Array(10), 2);"
`;

exports[`Array.prototype.fill should replace Array.prototype.fill with a polyfill 1`] = `
"import _Array_prototype_fill from \\"babel-plugin-transform-polyfills/polyfills/Array/prototype/fill\\";

_Array_prototype_fill(Array(10));"
`;
